### PR TITLE
Homework 7 jdi

### PR DESCRIPTION
**Pull request by AndreySinelnikov**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw7_jdi/JdiTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.epam.tc.hw7_jdi.entities.MetalsAndColorsEntry;
+import org.testng.annotations.Test;
+
+public class JdiTest implements TestsInit {
+
+    @Test(dataProvider = "testDataSet", dataProviderClass = JdiTestDataProvider.class)
+    public void testMetalAndColorsSubmission(MetalsAndColorsEntry metalsAndColorsEntry) {
+        // 1. Login on JDI site as User
+        open();
+        loginAs(ROMAN);
```
Comment (pnatashap) [inline src/test/java/com/epam/tc/hw7_jdi/JdiTest.java]: login is not a part of the test, it's only a precondition in this case

File: src/test/java/com/epam/tc/hw7_jdi/site/MetalsAndColorsPage.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    public static MetalsAndColorsForm metalsAndColorsForm;
+    @Css("#user-icon")
+    public static Icon userIcon;
+    @UI(".btn-login")
+    public static Button logoutButton;
+
+    @Css(".results li")
+    private static List<UIElement> resultTab;
+
+    public static void logout() {
```
Comment (pnatashap) [inline src/test/java/com/epam/tc/hw7_jdi/site/MetalsAndColorsPage.java]: A bit strange, that login and logout are located in different pages

**Review by pnatashap — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw7_jdi/JdiTest.java (hunk 1 above)
Comment (AndreySinelnikov) [reply src/test/java/com/epam/tc/hw7_jdi/JdiTest.java]: moved to BeforeMethod in TestsInit

File: src/test/java/com/epam/tc/hw7_jdi/site/MetalsAndColorsPage.java (hunk 2 above)
Comment (AndreySinelnikov) [reply src/test/java/com/epam/tc/hw7_jdi/site/MetalsAndColorsPage.java]: logout method moved to JdiSite

File: src/test/java/com/epam/tc/hw7_jdi/JdiTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        // Expected: Metals & Colors page is opened
+        metalsAndColorsPage.checkOpened();
+        // 3. Fill form Metals & Colors by JSON data
+        metalsAndColorsForm.fill(metalsAndColorsEntry);
+        metalsAndColorsForm.submit();
+        // 4. Result sections should contain data from JSON
+        assertThat(getResult())
+            .isEqualTo(metalsAndColorsEntry.getExpectedResult());
+
+        logout();
```
Comment (pnatashap) [inline src/test/java/com/epam/tc/hw7_jdi/JdiTest.java:28]: We do not need to logout for every test. Parametrized test is needed to check functionality, for this case login is a precondition

**Review by pnatashap — APPROVED** [review]

**Pull request merged**